### PR TITLE
Add Generic search

### PR DESCRIPTION
**Pull request by LanternNassi**
# Description

This is to add generic search 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: api_docs.yml [hunk 1]
```diff
@@ -3853,6 +3853,33 @@ paths:
...
+        - application/json
+      parameters:
+        - in: header
+          name: Authorization
+          required: true
+          type: string
+        - in: query
+          name: keywords
+          type: string
+          description: Kyewords to search
```
Comment (Mubangizi) [inline api_docs.yml]:
```suggestion
          description: Keywords to search
```

File: app/controllers/generic_search.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+class GenericSearchView(Resource):
+    @jwt_required
+    def get(self):
+        keywords = request.args.get('keywords', '')
+
+        #Schemas
+        projectSchema = ProjectListSchema(many=True)
+        
+        # projects 
+        projects = Project.query.filter(Project.name.ilike('%'+keywords+'%')).order_by(Project.date_created.desc())
```
Comment (Mubangizi) [inline app/controllers/generic_search.py]: This is nice, you need to add pagination then I think it should be good

Comment (LanternNassi) [reply app/controllers/generic_search.py]: Ok lemme add it

File: api_docs.yml [hunk 3]
```diff
@@ -3853,6 +3853,44 @@ paths:
...
+          required: true
+          type: string
+        - in: query
+          name: keywords
+          type: string
+          description: Keywords to search
+        - in: query
+          name: type
+          type: string
+          description: Enter search type (projects, apps, tags)
```
Comment (LanternNassi) [inline api_docs.yml:3874]:
```suggestion
          description: Enter search type (projects, users, tags)
```

**Pull request merged**